src/NHibernate/Util/ReflectHelperMemoryAssembly.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NHibernate.Util
+{
+	public class ReflectHelperMemoryAssembly
+    {
+		private static readonly ReflectHelperMemoryAssembly instance = new ReflectHelperMemoryAssembly();
+		private Dictionary<string, Assembly> _assemblyList;
+
+		public static ReflectHelperMemoryAssembly Instance
+		{
+			get
+			{
+				return instance;
+			}
+		}
+
+		/// <summary>
+		///  Remove all stored assembly in instance.
+		/// </summary>
+		public void Reset()
+		{
+			_assemblyList = new Dictionary<string, Assembly>();
+		}
+
+		public ReflectHelperMemoryAssembly()
+		{
+			Reset();
+		}
+
+		/// <summary>
+		///  Add new assembly to store.
+		/// </summary>
+		/// <param name="assembly">Assembly object.</param>
+		/// <returns>True if assembly was stored</returns>
+		public bool AddMemoryAssembly(Assembly assembly)
+		{
+			string name = assembly.FullName;
+			if (name.Length > 0 && !_assemblyList.ContainsKey(name))
+			{ 
+				_assemblyList.Add(name, assembly);
+				return true;
+			}
+			return false;
+		}
+
+		/// <summary>
+		///  Get number of stored assembly.
+		/// </summary>
+		public int GetAssemblyCount()
+		{
+			return _assemblyList.Count;
+		}
+
+		/// <summary>
+		///  Get type from specific assembly.
+		/// </summary>
+		/// <param name="type">Specify name class to get.</param>
+		/// /// <param name="assemblyName">Specify assembly name.</param>
+		public System.Type GetType(string type, string assemblyName)
+		{
+			return _assemblyList.ContainsKey(assemblyName)
+				?
+					_assemblyList[assemblyName].GetType(type)
```

Comment:
Use `TryGetValue` instead of `if contains then get`, once switched to concurrent.